OSDOCS-18979: Added instructions for creating multiple users with htpasswd (#111157)
openshift/openshift-docs: modules/config-htpasswd-idp-from-file.adoc

// Module included in the following assemblies:
//
// * authentication/sd-configuring-identity-providers.adoc

:_mod-docs-content-type: PROCEDURE

[id="config-htpasswd-idp-from-file_{context}"]

> = Configuring an htpasswd identity provider by using the file upload.

Reviewer: ```suggestion
= Configuring an htpasswd identity provider by using the file upload
```

[role="_abstract"]

> You can create an htpasswd identity provider by uploading a user file in the {cluster-manager} web user interface (UI).

Reviewer 2: So we don't do `htpasswd` identity?  or `htpasswd identity`? 

I trust you that we don't, it's just interesting that we don't, lol

[NOTE]
====

> To ensure that your htpasswd file is correctly hashed, you should use the link:https://docs.redhat.com/en/documentation/openshift_container_platform/4.21/html/authentication_and_authorization/configuring-identity-providers#creating-htpasswd-file[htpasswd tool] to create this file.

Reviewer 2: May we remove the "should", so that users simply do our instructions instead of thinking that they should? Maybe make the note more user focused while we're at it, something like: 

```suggestion
Ensure that you correctly hashed your htpasswd file by using the link:https://docs.redhat.com/en/documentation/openshift_container_platform/4.21/html/authentication_and_authorization/configuring-identity-providers#creating-htpasswd-file[htpasswd tool] to create this file.
```

Reviewer 2: If you don't want this whole rewrite, even something simpler like:

```suggestion
To ensure that your htpasswd file is correctly hashed, use the link:https://docs.redhat.com/en/documentation/openshift_container_platform/4.21/html/authentication_and_authorization/configuring-identity-providers#creating-htpasswd-file[htpasswd tool] to create this file.
```

====

.Procedure

. Select your cluster from the *Cluster List* page on {cluster-manager-url}.

. Select *Access control* -> *Identity providers*.

. Click *Add identity provider*.

. Select *htpasswd* from the *Identity Provider* list.

. Add a unique name in the *Name* field for the identity provider.

. Select *Upload an htpasswd file*.

. In the *htpasswd file* field, drag your configured htpasswd file or select *Browse* to locate the file on your local drive.

> . Select *Add* to create the htpasswd identity provider and the users within the uploaded file.

Reviewer: ```suggestion
. Select *Add* to create the htpasswd identity provider and add the users from the uploaded file.
```

.Verification

> * The configured htpasswd identity provider is visible on the *Access control* -> *Identity providers* page.

Reviewer 2: I love verification steps! I'm used to them being mnore a step the user does though, something like: 

```suggestion
* You can see your configured htpasswd identity provider on the *Access control* -> *Identity providers* page.
```

Reviewer 2: or 
```suggestion
* Esnure that your htpasswd identity provider is visible on the *Access control* -> *Identity providers* page.
```
